project-codeflare/codeflare-sdk
# Update Dependencies: Include 'ray[data]' for PyArrow Support

# Issue link


Resolves #423 

# What changes have been made

Removed the individual PyArrow dependency, where instead we are making use of the data extra i.e., `"ray[data]"`, which includes the required dependency. This is to avoid any future potential incompatibility.

# Verification steps



- Checkout this PR
- Install codeflare-sdk in your kubernetes cluster. Probably easiest way is to run `poetry build`, then install the generated wheel with `pip install codeflare_sdk-0.0.0.dev0-py3-none-any.whl`.
- Clone the codeflare-sdk repository.
- Run through the guided-demo notebook `3_basic_interactive.ipynb` where we will set in our ClusterConfiguration instascale to false and GPU to 0 as scaling doesn't occur before the error is encountered. Run the demo until the point where we run `ray.init` and make sure the Ray cluster is up and running.
- No errors encountered.

I have tested this change as well on the `2_basic_jobs.ipynb` notebook to verify dependencies work as expected. 

## Checks
- Testing Strategy
